src/Form/Control/ScopeBuilder.php

```diff
@@ -316,11 +343,40 @@ public function setModel(Model $model)
     {
         $model = parent::setModel($model);
 
+        if (!$this->dataCb) {
+            $this->dataCb = Callback::addTo($this);
+        }
+        $this->dataCb->set([$this, 'outputApiResponse']);
+
         $this->buildQuery($model);
 
         return $model;
     }
 
+    /**
+     * Output lookup search query data.
+     */
+    public function outputApiResponse()
+    {
+        $fieldName = $_GET['atk_vlookup_field'] ?? null;
+        $query = $_GET['atk_vlookup_q'] ?? null;
+        $data = [];
+        if ($fieldName) {
+            $model = $this->model->getField($fieldName)->reference->refModel();
+            if (!empty($query)) {
+                $model->addCondition($model->title_field, 'like', '%' . $query . '%');
```

Comment:
this is quite fragile query and it needs much more code to be really universal, like to work correctly across collations - can we instance a standard Lookup here are reuse code for data query from there?

Reply:
not sure what you mean here. This is basically how all other components are doing. Can you explain a bit more?

Reply:
I think he means that it should reuse something we already have implemented in Lookup field. But I guess it was not in lookup field, but in QuickSearch.

Reply:
both, Lookup and QuickSearch use` $model->addCondition($model->title_field, 'like', '%' . $query . '%');`

Reply:
ok with me for now